### PR TITLE
Websql

### PR DESCRIPTION
**Pull request by dereke**
This implements a websql driver.
You can either use it with the 'websql' npm module (which is just like using sqlite3 in node) or supply the driver with the browsers `window.openDatabase` function.

### CODE REVIEW

File: test/websqlSpec.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+var config = {
+  driver: "websql",
+  config: { filename: __dirname + "/test.db" }
+};
+
+describeDatabase("websql", config, database, function () {
+  describe('connection', function () {
+    it('can accept a file: URL', function () {
+      var db = sworm.db('file://' + config.config.filename + '?asdf=asdf');
```
Comment (refractalize) [inline test/websqlSpec.js]: is this just going to load the `sqlite` driver?

File: test/websqlSpec.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      return db.query('select * from people')
+    });
+
+    it('can accept a filename', function () {
+      var db = sworm.db(config.config.filename);
```
Comment (refractalize) [inline test/websqlSpec.js]: same as this

Comment (dereke) [inline test/websqlSpec.js]: hmm I guess it will, the test is a bit redundant isn't it?
remove the test? or should we support `websql://somefile.db`

Comment (refractalize) [inline test/websqlSpec.js]: yeah just remove i think

**Comment by refractalize** [comment]
thanks @dereke, in v3.5.0

**Pull request merged**